packages/flutter/test/widgets/text_scaler_backward_compatibility_test.dart

```diff
@@ -0,0 +1,267 @@
+      );
+      expect(renderObject.textScaleFactor, 1.0);
+
+      renderObject.textScaleFactor = 3.0;
+      expect(renderObject.textScaleFactor, 3.0);
+      expect(renderObject.textScaler, const TextScaler.linear(3.0));
+
+      renderObject.textScaler = const TextScaler.linear(4.0);
+      expect(renderObject.textScaleFactor, 4.0);
+    });
+
+    test('RenderParagraph', () {
+      final RenderParagraph renderObject = RenderParagraph(
+        const TextSpan(
+          text: 'test',
+          style: TextStyle(height: 1.0, fontSize: 10.0),
+        ),
+        textDirection: TextDirection.ltr,
+      );
+      expect(renderObject.textScaleFactor, 1.0);
+
+      renderObject.textScaleFactor = 3.0;
+      expect(renderObject.textScaleFactor, 3.0);
+      expect(renderObject.textScaler, const TextScaler.linear(3.0));
+
+      renderObject.textScaler = const TextScaler.linear(4.0);
+      expect(renderObject.textScaleFactor, 4.0);
+    });
+  });
+
+  group('Widgets backward compatibility', () {
+    testWidgets('RichText', (WidgetTester tester) async {
+      await tester.pumpWidget(
+        RichText(
+          textDirection: TextDirection.ltr,
+          text: const TextSpan(),
+          textScaleFactor: 2.0,
+        ),
+      );
+
+      expect(
+        tester.renderObject<RenderParagraph>(find.byType(RichText)).textScaler,
+        const TextScaler.linear(2.0),
+      );
+      expect(tester.renderObject<RenderParagraph>(find.byType(RichText)).textScaleFactor, 2.0);
+    });
+
+    testWidgets('Text', (WidgetTester tester) async {
+      await tester.pumpWidget(
+        const Text(
+          'text',
+          textDirection: TextDirection.ltr,
+          textScaleFactor: 2.0,
+        ),
+      );
+
+      expect(
+        tester.renderObject<RenderParagraph>(find.text('text')).textScaler,
+        const TextScaler.linear(2.0),
+      );
+    });
+
+    testWidgets('EditableText', (WidgetTester tester) async {
+      final TextEditingController controller = TextEditingController();
+      final FocusNode focusNode = FocusNode(debugLabel: 'EditableText Node');
+      const TextStyle textStyle = TextStyle();
+      const Color cursorColor = Color.fromARGB(0xFF, 0xFF, 0x00, 0x00);
+      await tester.pumpWidget(
+        MediaQuery(
+          data: const MediaQueryData(),
+          child: Directionality(
+            textDirection: TextDirection.rtl,
+            child: EditableText(
+              backgroundCursorColor: cursorColor,
+              controller: controller,
+              focusNode: focusNode,
+              style: textStyle,
+              cursorColor: cursorColor,
+              textScaleFactor: 2.0,
+            ),
+          ),
+        ),
+      );
+
+      final RenderEditable renderEditable = tester.allRenderObjects.whereType<RenderEditable>().first;
+      expect(
+        renderEditable.textScaler,
+        const TextScaler.linear(2.0),
+      );
+    });
+  });
```

Comment:
Thanks for writing these backwards compatibility tests even though they're short lived!